model/src/main/java/io/paradoxical/cassieq/model/auth/SignedRequestParameters.java

```diff
@@ -0,0 +1,82 @@
+package io.paradoxical.cassieq.model.auth;
+
+import com.godaddy.logging.Logger;
+import com.google.common.base.CharMatcher;
+import com.google.common.base.Joiner;
+import io.paradoxical.cassieq.model.accounts.AccountKey;
+import io.paradoxical.cassieq.model.accounts.AccountName;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import javax.validation.constraints.NotNull;
+import java.util.Base64;
+import java.util.EnumSet;
+
+import static com.godaddy.logging.LoggerFactory.getLogger;
+
+@Value
+@Builder
+public class SignedRequestParameters implements RequestParameters {
+
+    private static final Logger logger = getLogger(SignedRequestParameters.class);
+
+    @NonNull
+    @NotNull
+    private final AccountName accountName;
+
+    @NonNull
+    @NotNull
+    private final String requestMethod;
+
+    @NonNull
+    @NotNull
+    private final String requestPath;
+
+    private final String providedSignature;
+
+    @Override
+    public EnumSet<AuthorizationLevel> getAuthorizationLevels() {
+        return AuthorizationLevel.All;
+    }
+
+    @Override
+    public boolean verify(final AccountKey key) throws Exception {
+        final Mac hmacSHA256 = Mac.getInstance("HmacSHA256");
```

Comment:
do we want to put verify in a utility or something? it looks the same in sign url params